src/ExplorerComponent.ts

```diff
@@ -1,43 +1,172 @@
+                                that.gotoBreadcrumb(self.data);
+                            });
+                    },
+                    template: $.templates.breadcrumbTemplate
+                },
+                item: {
+                    init: function (tagCtx, linkCtx, ctx) {
+                        this.data = tagCtx.view.data;
+                        this.data.parents
+                    },
+                    onAfterLink: function () {
+                        var self: any = this;
+
+                        self.contents('.explorer-item')
+                            .on('click', 'a.explorer-folder-link', function() {
+                                that.openFolder(self.data);
+                            })
+                            .on('click', 'a.explorer-item-link', function() {
+                                that._emit(ExplorerComponent.Events.EXPLORER_NODE_SELECTED, self.data);
+                            });
+                    },
+                    template: $.templates.itemTemplate
+                }
+            });
 
             return success;
         }
-        
+
+        protected _draw(): void {
+            this._$view.link($.templates.pageTemplate, { parents: this._parents, current: this._current });
+        }
+
+        protected _sortCollectionsFirst(a: Manifold.ITreeNode, b: Manifold.ITreeNode): number {
+            if (a.data.type === b.data.type) {
+                // Alphabetical
+                return a.label < b.label ? -1 : 1;
+            }
+            // Collections first
+            return b.data.type.indexOf('collection') - a.data.type.indexOf('collection');
+        }
+
+        public gotoBreadcrumb(node: Manifold.ITreeNode): void {
+            let index: number = this._parents.indexOf(node);
+            this._current = this._parents[index];
+            this._parents = this._parents.slice(0, index + 1);
+            this._draw();
+        }
+
+        protected _switchToFolder(node: Manifold.ITreeNode): void {
+            node.nodes.sort(this._sortCollectionsFirst);
+            this._parents.push(node);
+            this._current = node;
+            this._draw();
+        }
+
+        public openFolder(node: Manifold.ITreeNode): void {
+            if (!node.data.isLoaded) {
+                let that = this;
+                node.data.load().then(function (collection: Manifesto.Collection): void {
+                    console.log(collection);
+                    node.nodes = collection.members.map(function (op: Manifesto.IIIFResource): Manifesto.TreeNode {
+                        // TODO: OFFICIAL CONVERSION MUST EXIST
+                        let data: any = op;
+                        data.type = op.__jsonld['@type'].toLowerCase();
+                        let treeNode: Manifesto.TreeNode = new Manifesto.TreeNode(op.__jsonld.label, data);
+                        treeNode.parentNode = node;
+                        return treeNode;
```

Comment:
I'm concerned about this part of the code, I have yet to find a better, pre-existing way to convert IIIFResources to TreeNodes. It must exist.